### ISSUE TITLE
Controller Fezzes And TouchID Number Gets Messed Up (C2 RunTime)

### ISSUE THREAD

**Issue by Tarek200** [Scirra/Construct-3-bugs#3400]
**On iPhones:**
The longer it takes the layout to load the more the controller gets freeze randomly and the TouchID number gets messed up instead off given (ID = 0) for the first touch it gives long numbers like **(-503659549)**
Once it gets freeze there is no way to unblocked it, gets stuck on the same position even if you change layouts and come back to the same layout it will be still stuck on that position (X,Y) and the only way to unblocked is resetting the iPhone.

**Note:**
The Freeze on iPhones: it will be hard to reproduce as it happens very random 
TouchID: you can reproduce it any Time 

**On Ipads:**
Is the same problem as described above but the only differences are:
The Freeze: you can reproduce it any Time as it freezes 90% of the times
TouchID: you can reproduce it any Time 

**C2 RunTime .c3p**
[c3p](https://www.dropbox.com/s/7a5s3ccv1ulfmx1/controller%20bug1.c3p?dl=0)

## Steps to reproduce
**Note:** 
I never misusing the TileMap like that, you know big image for small tiles etc..., it is only to recreate the bug purpose.
Also for more high chances to reproduce the "controller freeze"  bug is better to use IPad as a testing device, example Ipad Pro 11 inch

1. Remote Preview the project from the first layout "Menu" Layout 
2. Click the Green Square and keep tapping on it as much as you can until it loads the next layout 
3. Once the next layout  "Controllers" is loaded then touch anywhere on the screen

## Observed result

1-Controller gets freeze
2-TouchID shows incorrect value >>> example: (-503659549)
3-Even if you change the layout and come back the controller it will be still stuck and freeze

## Expected result
1-Controller will not get freeze
2-TouchID to show correct value >>> example: First touch = 0

## More details
**Tested on fisical Devices:**
-iPhone: 6, 7, 8 plus, XS Max
-Ipad: Pro 11 inch

**Notes:**
Looks like the longer c2 Loads the layout and the bigger is the screen the more often the controller will freeze and it happens only on the devices either by previewing over Wifi or either by exporting it and installing it on the Device.
Example on the Pc it doesn't happen and also on C3RunTime it doesn't happen either, this only happens on (C2Runtime and C2)

Its been happening at least for the last two years I don't remember all the Construct versions I tested

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 77.0.3865.90
Browser engine: Chromium
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36
C3 release: r170 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 13 gb
Storage usage (approx): 376 mb (2.7%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 1070 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_filter_anisotropic
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
This is a difficult issue to interpret:

- what do you mean by "controller"? Typically that refers to a gamepad, but then you go on to talk about touch input
- touch IDs are not guaranteed to be any particular numbers, so the fact you get a large or negative number is irrelevant, so long as every touch has a different ID
- each bug report should only describe one issue to avoid confusion (such as here)
- you have not included information about the iOS version

Issue #2750 sounds similar and only affects iOS 12 due to a bug in iOS itself. It's fixed in iOS 13. Therefore I would assume this is a duplicate of that. Please always follow the bug report guidelines to ensure your report is clear and useful.

**Comment by Tarek200** [comment]
So hard to report just a simple bug, yes the Gamepad it freezes and I included the touch id problem as it may be related as both issues happen at the same time, no is not the same bug as the one you linked, I just wasted 3 hours to report a simple bug for nothing, I have to open a new bug report now.

**Comment by AshleyScirra** [comment]
It's essential to write bug reports clearly so they can be understood. Please take care to ensure it is clear and unambiguous what you are reporting, otherwise it becomes incredibly confusing to deal with the report.
